Dockerfile

```diff
@@ -1,19 +1,16 @@
 FROM node:8-alpine
 
-ARG VERSION
-ARG TYPE=basic
+LABEL maintainer Vincenzo Chianese, vincenzo@express-gateway.io
 
+ENV EG_VERSION 1.7.1
 ENV NODE_ENV production
-# this will enable polling, hot-reload will work on docker or network volumes
+# Enable chokidar polling so hot-reload mechanism can work on docker or network volumes
 ENV CHOKIDAR_USEPOLLING true
 
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
+RUN npm install express-gateway@$EG_VERSION
 
-RUN npm install express-gateway@$VERSION && \
-    ./node_modules/.bin/eg gateway create -n gateway -d . -t $TYPE && \
-    npm cache clean --force
+WORKDIR /node_modules/express-gateway
```

Comment:
This is pretty bizarre.  Why do this over `eg gateway create`?  Or it might be possible to just copy the `basic` template and use that as a starting point.

Reply:
I analyzed different docker official images based on npm packages and it seems to be the standard. In particular:

1. You `npm install -g package-name` and then you start the package through the cli (that's something we do not have)

2. You install the package locally and you put yourself in the directory where the relevant files are. See [here for an example](https://github.com/mongo-express/mongo-express-docker/blob/master/Dockerfile#L33). We can put the default template as part of the configuration, though.

Our scaffolded mechanism won't be accepted — you'd have two gateway copies.

Reply:
What about doing 

```
COPY ./bin/generators/gateway/templates/basic/* /usr/src/app
WORKDIR /usr/src/app
RUN npm install express-gateway@$EG_VERSION
```

Thoughts?

Reply:
I think that can work (if they do not force us to have the docker image in a separate file).

The basic template might need to be modified though because:

1. The admin API is listening on `localhost` only — so it won't be accessible outside the container.

2. The template is configuring the gateway to do basically nothing — so if anybody tries to send a request, they'll receive a `404`. I'm totally ok with that, I just wanted to make you guys aware of it.

Reply:
> It turns out that if you scaffold a gateway, the hardcoded `configPath` takes the precedence [over the environment variable](https://github.com/ExpressGateway/express-gateway/blob/master/lib/index.js#L20-L21). We need to invert this.

@DrMegavolt Maybe you know/recall the reason?

Reply:
I don't know why. +1 to making some generated files to run, not entire EG module. it should be possible to override server.js file as well in this case by mounting volume